### ISSUE TITLE
404 Not Found has not been implemented

### ISSUE THREAD

**Issue by squgeim** [mesaugat/express-api-es6-starter#22]
In `src/index.js`, the last middleware to fallback to is `methodNotAllowed`. This throws a `405`. Shouldn't the fallback be `404`, at least for a `GET` request? There already is a `notFoundError` implemented in `middlewares/errorHandler`.

**Comment by mesaugat** [comment]
I changed it to `405`. I have found people use `Method Not Allowed` when there is no resource/route for something.

Most people seem (at least beginners) to confuse themselves of **'Resource being not found'** when `404` is shown. `405` is more meaningful for cases like those. Yes, we can use `404` and replace it with a message like `Route not found` because only having a message like `Not found` causes some confusion.
